COMPILE.md

```diff
@@ -1,92 +1,69 @@
-
-The model in Tengine can run faster than the C++ source code here because Tengine has been optimized according to ARM CPU. There are detailed manual and example at Tengine web site: https://github.com/OAID/Tengine/tree/master/examples/YuFaceDetectNet
-
-## Cross build for aarch64
-
-1. Set cross compiler for aarch64 (please refer to aarch64-toolchain.cmake)
-2. Set opencv path since the example code depends on opencv
-
-```bash
-cmake \
-    -DENABLE_NEON=ON \
-    -DCMAKE_BUILD_TYPE=RELEASE \
-    -DCMAKE_TOOLCHAIN_FILE=../aarch64-toolchain.cmake \
-     ..
-
-make
-```
-
-## Native build for avx2
-
-```bash
-cmake \
-    -DENABLE_AVX2=ON \
-    -DCMAKE_BUILD_TYPE=RELEASE \
-    -DDEMO=ON \
-     ..
-
-make
-```
-
-## Android platform compilation instructions
-
-+ Install ndk
-  - Download and install to /home/android-ndk from https://developer.android.com/ndk/downloads 
-  - Setting environment variables
+### Table of contents:
+[Compilation](#compilation)
+  - Windows 10
+     - [with Visual Studio 2019](#windows-10-with-visual-studio-2019)
+     - [with MinGW](#windows-10-with-mingw)
+  - [Linux/Ubuntu](#linux-or-ubuntu)
+  - [Android](#android)
+  - [OpenCV DNN](#opencv-dnn)
+  - [Cross build for aarch64](#cross-build-for-aarch64)
+
+[Usage](#usage)
+
+[Example](#example)
+
+## Compilation
+To use the library, you can do either of the following:
+- Copy the .cpp files in src directly to your project source folder, and compile them with the other files in your project.
+      The source code are written in standard C/C++, so they should compile on any platform that supports C/C++;
+- Or follow the steps below to generate dynamic/static libraries under different environments.
+### Windows 10 with Visual Studio 2019
+  0. Set up OpenCV with 4.51+ version.
+  1. Download libfacedetection and then run powershell terminal <ins>as administrator</ins>:
+
+            cd libfacedetection
+            mkdir build
+            cd build
+            cmake .. -DCMAKE_INSTALL_PREFIX=install -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Release -DDEMO=OFF
+            cmake --build . --config Release
+            cmake --build . --config Release --target install
+
+      Dynamic library(facedetection.dll) is generated. Then, to generate static library(lib), you need to set the parameter `BUILD_SHARED_LIBS` to `OFF` with the above commands.
+  2. In your console application property pages, <ins>under Release mode</ins> (because the build type is Release), add the path of your `libfacedetection\build\install\include\facedetection` to VC++ Directories -> Include Directories and the path of your `libfacedetection\build\install\lib` to VC++ Directories -> Library Directories, and add `facedetection.lib` to Linker -> Input -> Additional Dependencies.
+
+  3. Add `#include "facedetectcnn.h"` to your source files to use the libraries.
```

Comment:
Can you add a code example explaining how to use .dll and .lib in Visual Studio? If it is compatible with `examples/detect-image.cpp`, just offer a directory structure tree like the following to help people getting started:
```shell
PS C:\compile-test> tree
Folder PATH listing
Volume serial number is ***
C:.
├─detect-image.cpp
├─facedetect.h
└─facedetection.dll ( or facedetection.lib)

```